### ISSUE TITLE
Confirm Only After Add?

### ISSUE THREAD

**Issue by dan1982code** [nvdaes/clipContentsDesigner#7]
Hi,

Confirm to emulate copy/cut are useful features, but presently bogged down by happening every time ctrl+c or ctrl+x is used.

I think that the confirmation box should only show up if win+NVDA+c has been used at least once since the previous copy/cut. Otherwise, there seems no reason to confirm ctrl+c or ctrl+x. i.e. the warning should warn people that they're about to overwrite the clipboard that they've been incrementally building; otherwise, overwriting the clipboard is expected behaviour whether this addon is being used or not.

 Thank you.

**Comment by nvdaes** [comment]
> I think that the confirmation box should only show up if win+NVDA+c has been used at least once since the previous copy/cut.

If you want this behavior, you can check the Text in clipboard to confirm checkbox.
- The clipboard can contain text added after pressing control+c, or by other add-on, etc, and plugins can be reloaded or NVDA could be restarted, so we can distinguish accurately if the text has been added by this add-on or not, and the important thing is if the text is not present, not the origin.
- Even when overriding the clipboard contents is the expected behavior, since confirmation can be useful for this add-on and this can prevent overriding the clipboard when we have copied a file, not just text, the option can be used even when the previous related to text contained in clipboard checkbox is unchecked.

**Comment by dan1982code** [comment]
Interesting! But then we get a confirmation box on every single use of ctrl+c or ctrl+x, whether we're currently using the functionality of the addon or not.

Could we have a "confirm only after add" checkbox, which implements what I have suggested? i.e. ctrl+c or ctrl+x gives the confirmation box only if a win+NVDA+c has occurred since the last copy or cut?

I see that it's flawed, for reasons you mentioned, plus if people use edit|copy or other ways to manipulate the clipboard. But my suggestion does fix the annoying case of ctrl+c giving the confirmation every time, while preventing the loss of building up a clipboard with win+NVDA+c due to an inadvertent ctrl+c.

Thank you.

**Comment by nvdaes** [comment]
I think this is not possible for the mentioned reasons.

**Comment by nvdaes** [comment]
I will close this issue. You can reopen it or create a new issue for further comments. Thanks for your valuable feedback!
